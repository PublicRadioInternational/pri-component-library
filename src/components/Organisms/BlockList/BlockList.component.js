/**
 * @file BlockList.component.js
 * Exports a block list component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './BlockList.css';

/**
 * Component that renders a Block List.
 */
const BlockList = ({ title, children }) => (
  <div className={styles.blockList}>
    {title && <h2 className={styles.title}>{title}</h2>}
    <div className={styles.list}>{children}</div>
  </div>
);

BlockList.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node
};

BlockList.defaultProps = {
  title: null,
  children: []
};

export default BlockList;
