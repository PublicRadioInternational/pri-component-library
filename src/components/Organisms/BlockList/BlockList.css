/* import colors... */
@value colors: "../../00_global/colors.css";
@value whiteDark, grayDark from colors;

.blockList {
  background: whiteDark;
  margin-bottom: 1rem;
}

.title {
  border-bottom: 1px dotted #ddd;
  padding: 0.75rem;
  font-size: 1.1rem;
  font-weight: normal;
}

.list > * {
  border-top: 1px dotted #ddd;
  counter-increment: my-counter;
}

.list > *::before {
  background: #ddd;
  border-radius: 50%;
  color: grayDark;
  content: counter(my-counter);
  display: block;
  font-size: 1rem;
  height: 20px;
  line-height: 1.2;
  margin: 0 auto 3px;
  width: 20px;
}

.list > *:first-child {
  border-top: none;
}
