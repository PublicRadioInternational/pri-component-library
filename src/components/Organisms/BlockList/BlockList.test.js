/**
 * @file BlockList.test.js
 * Contains tests for BlockList.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import BlockList from './BlockList.component';

describe('<BlockList />', () => {
  it('Matches the Block List snapshot', () => {
    const component = renderer
      .create(<BlockList title="Three Ways to Support PRI" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
