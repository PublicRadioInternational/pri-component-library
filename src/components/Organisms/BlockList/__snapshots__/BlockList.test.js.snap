// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockList /> Matches the Block List snapshot 1`] = `
<div
  className="blockList"
>
  <h2
    className="title"
  >
    Three Ways to Support PRI
  </h2>
  <div
    className="list"
  />
</div>
`;
